alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt

```diff
@@ -582,6 +582,116 @@ class MonitorRunnerIT : AlertingRestTestCase() {
+                params = mapOf(),
+                url = "",
+                connection_timeout = 5,
+                socket_timeout = 5)
+        val monitor = createMonitor(randomMonitor(inputs = listOf(input)))
+        val response = executeMonitor(monitor.id)
+
+        val output = entityAsMap(response)
+        @Suppress("UNCHECKED_CAST")
+        val inputResults = output.stringMap("input_results")
+        val resultsContent = (inputResults?.get("results") as ArrayList<*>).get(0)
+        val errorMessage = inputResults.get("error")
+
+        assertEquals(monitor.name, output["monitor_name"])
+        @Suppress("UNCHECKED_CAST")
+        assertTrue("Monitor results should contain cluster_name field, but was actually: $resultsContent",
+                resultsContent.toString().contains("cluster_name"))
+        @Suppress("UNCHECKED_CAST")
+        assertTrue("Error message should not exist, error: $errorMessage", errorMessage == null)
+    }
+
+    fun `test monitor HttpInput with alert triggered`() {
+        putAlertMappings() // Required as we do not have a create alert API.
+        val trigger = randomTrigger(condition = Script("""
+            return ctx.results[0].number_of_pending_tasks < 1
+        """.trimIndent()), destinationId = createDestination().id)
+        val clusterIndex = randomInt(clusterHosts.size - 1)
+        val input = randomHttpInput(
+                scheme = clusterHosts[clusterIndex].schemeName,
+                host = clusterHosts[clusterIndex].hostName,
+                port = clusterHosts[clusterIndex].port,
+                path = "_cluster/health",
+                params = mapOf(),
+                url = "",
+                connection_timeout = 5,
+                socket_timeout = 5)
+        val monitor = createMonitor(randomMonitor(inputs = listOf(input), triggers = listOf(trigger)))
+        val response = executeMonitor(monitor.id)
+
+        val output = entityAsMap(response)
+        assertEquals(monitor.name, output["monitor_name"])
+        for (triggerResult in output.objectMap("trigger_results").values) {
+            assertTrue("Should be triggered.", triggerResult.objectMap("action_results").isNotEmpty())
+        }
+
+        val alerts = searchAlerts(monitor)
+        assertEquals("Alert not saved, $output", 1, alerts.size)
+        verifyAlert(alerts.single(), monitor, ACTIVE)
+    }
+
+    fun `test monitor HttpInput with no alert triggered`() {
+        putAlertMappings() // Required as we do not have a create alert API.
+        val trigger = randomTrigger(condition = Script("""
+            return ctx.results[0].status.equals("red")
+        """.trimIndent()))
+        val clusterIndex = randomInt(clusterHosts.size - 1)
+        val input = randomHttpInput(
+                scheme = clusterHosts[clusterIndex].schemeName,
+                host = clusterHosts[clusterIndex].hostName,
+                port = clusterHosts[clusterIndex].port,
+                path = "_cluster/health",
```

Comment:
Need a test with a plain text or other form of non-JSON output.

Reply:
Also included in another comment: I have this [test](https://github.com/ann3431/alerting/blob/005b5c2e3236f945a5207453ccf31d18bd56044b/alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt#L585) in `MonitorRunnerIT` that uses the _cat/indices API which returns plain text. Is this sufficient? Thanks!